geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheEnvironment.java

```diff
@@ -0,0 +1,134 @@
+package org.geowebcache;
+
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.PlaceholderConfigurerSupport;
+import org.springframework.core.Constants;
+import org.springframework.util.PropertyPlaceholderHelper;
+import org.springframework.util.PropertyPlaceholderHelper.PlaceholderResolver;
+
+/**
+ * Utility class uses to process GeoWebCache configuration workflow through external environment variables.
+ * 
+ * This class must be used everytime we need to resolve a configuration placeholder at runtime.
+ * <p>
+ * An instance of this class needs to be registered in spring context as follows.
+ * 
+ * <pre>
+ * <code>
+ *         &lt;bean id="geoWebCacheEnvironment" class="org.geowebcache.GeoWebCacheEnvironment" depends-on="geoWebCacheExtensions"/&gt;
+ * </code>
+ * </pre>
+ * 
+ * It must be a singleton, and must not be loaded lazily. Furthermore, this bean must be loaded before any beans that use it.
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class GeoWebCacheEnvironment {
+
+    /**
+     * logger
+     */
+    public static Log LOGGER = LogFactory.getLog(GeoWebCacheEnvironment.class);
+
+    private static final Constants constants = new Constants(PlaceholderConfigurerSupport.class);
+
+    /**
+     * Constant set via System Environment in order to instruct GeoWebCache to make use or not of the config placeholders translation.
+     * 
+     * Default to FALSE
+     */
+    public final static boolean ALLOW_ENV_PARAMETRIZATION = Boolean
```

Comment:
Is it possible to use GeoWebCacheExtensions.getProperty instead, to allow reading the conf from system, environment or servlet context variable?